server/servercontext.py

```diff
@@ -12,10 +13,11 @@ class ServerContext:
     Base class for managing connections and holding state about them.
     """
 
-    def __init__(self, connection_factory, name='Unknown server'):
+    def __init__(self, connection_factory, throttler, name='Unknown server'):
```

Comment:
Maybe `throttler=None` could be the default, then you don't need to explicitly pass `None` in the integration tests.